src/ostree/ot-admin-builtin-pin.c

```diff
@@ -28,9 +28,74 @@
 
 static gboolean opt_unpin;
 
+typedef enum
+{
+  OSTREE_DEPLOYMENT_INDEX = 0,
+  OSTREE_DEPLOYMENT_BOOTED = 1,
+  OSTREE_DEPLOYMENT_PENDING = 2,
+  OSTREE_DEPLOYMENT_ROLLBACK = 3
+} OstreeDeploymentType;
+
 static GOptionEntry options[]
     = { { "unpin", 'u', 0, G_OPTION_ARG_NONE, &opt_unpin, "Unset pin", NULL }, { NULL } };
 
+static gint64
+get_deployment_index (const OstreeDeployment *target_deployment, const GPtrArray *deployments)
```

Comment:
This is just `ostree_deployment_get_index`

Reply:
SGTM

Reply:
I should call this `ostree_match_deployment`, thanks for pointing this out, there's another very similarly named function...

Reply:
I see what you mean now, will use that function